src/logger.cpp

```diff
@@ -0,0 +1,87 @@
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <locale>
+#include <sstream>
+
+#include "logger.h"
+
+namespace ohl::logger {
+
+const char* logPath = "ohl.log";
+
+/**
+ * @brief Logs a logMessage to the log file ohl::logger::logPath. \n will be
+ * added at the end
+ *
+ * @param logMessage the message to be logged.
+ */
+void log( const char * logMessage ) {
+    // Open a log file for appending each time
+    // at least it deals with the flushing issue.
+    std::ofstream logFile( logPath, std::ios_base::app );
+    logFile << logMessage << "\n";
+    logFile.close();
+}
+/**
+ * @brief Logs a logMessage to the log file ohl::logger::logPath. \n will be
+ * added at the end
+ *
+ * @param logMessage the message to be logged.
+ */
+void log( const wchar_t * logMessage ) {
+    // Open a log file for appending each time
+    // at least it deals with the flushing issue.
+    std::wofstream logFile( logPath, std::ios_base::app );
+    logFile << logMessage << "\n";
+    logFile.close();
+}
+void log( const std::ostream& logMessage ) {
+    // Open a log file for appending each time
+    // at least it deals with the flushing issue.
+    std::ofstream logFile( logPath, std::ios_base::app );
+    logFile << logMessage.rdbuf() << std::endl;
+    logFile.close();
+}
+void log( const std::wostream& logMessage ) {
+    // Open a log file for appending each time
+    // at least it deals with the flushing issue.
+    std::wofstream logFile( logPath, std::ios_base::app );
+    logFile << logMessage.rdbuf() << "\n";
+    logFile.close();
+}
+/**
+ * @brief Logs AND prints a logMessage to the log file ohl::logger::logPath. \n will be
+ * added at the end
+ *
+ * @param logMessage the message to be logged.
+ */
+void logPrint( const char * logMessage ) {
+    std::cout << logMessage << "\n";
```

Comment:
Should append the `[OHL] ` prefix here too - it's debatable if it needs to be in the file, but definitely needs to be in every stdout message.